test/loopback-dl.test.js

```diff
@@ -1421,6 +1421,44 @@ describe('DataAccessObject', function() {
     assert.deepEqual(where, {or: [{age: 10}, {vip: true}]});
   });
 
+  it('should be able to coerce where clauses with array-like objects', function() {
+    where = model._coerce({scores: {0: '10', 1: '20'}});
+    assert.deepEqual(where, {scores: [10, 20]});
+
```

Comment:
One assert per test please. Split this test into multiple smaller tests, each test calling `_coerce` + `assert` only once.

(I do realise the existing code does not follow this best practice.)